myning/utils/utils.py

```diff
@@ -38,6 +38,37 @@ def get_random_array_item(arr: ListType[T]) -> T:
     return get_random_array_item_and_index(arr)[0]
 
 
+def boosted_random_choice(
+    arr: list[T], selector: Callable[[T], bool], percent_boost: float = 0.0
+) -> Optional[T]:
```

Comment:
```suggestion
) -> T | None:
```

I have a preference for the union type operator that I'd like to share with you 😆 

`Optional[T]` sounds more like the thing to use when defining a dataclass with optional properties

Reply:
I never thought of this! Thanks RJ for the enlightenment!